### PR TITLE
Feature chapter0

### PR DESCRIPTION
**Pull request by ThierryS72**
Features Chapter0 + new sounds and musics

### CODE REVIEW

File: Assets/Scripts/Chapters/ControllerChapter0.cs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        while (i < headCount -3)
+        {
+            heads.GetChild(i).gameObject.SetActive(i==randomVal);
+            i++;
+        }
+        */
+        selectedCharacterIndex = 10; //Otarie as head default
+        UpdateCharacterSelectionUI();
+    }
+    private void Update()
```
Comment (DoubleSentinel) [inline Assets/Scripts/Chapters/ControllerChapter0.cs]: Remove this whole method

File: Assets/Scripts/Chapters/ControllerChapter0.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        {
+            int index = Random.Range(0, listHeads.Count);
+            controllers.GetComponent<TestingEnvironment>().Characters.Add($"Objective{i}", listHeads[index]);
+            listHeads.RemoveAt(index); //Adapt listHeads
+        }
+    }
+
+    private void UpdateCharacterSelectionUI()
+    {
+        characterName.text = scenePlayerCh0.gameObject.transform.GetChild(1).GetChild(2).GetChild(selectedCharacterIndex).gameObject.name;
```
Comment (DoubleSentinel) [inline Assets/Scripts/Chapters/ControllerChapter0.cs]: change to `heads`

File: Assets/Scripts/Chapters/ControllerChapter0.cs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    //Load Ctreate Account View
+    public void LoadAccountCreation()
+    {
+        GameEventMessage.SendEvent("LoadCreateAccount");
+    }
+
+    [System.Serializable]
+    public class CharacterSelectObject
```
Comment (DoubleSentinel) [inline Assets/Scripts/Chapters/ControllerChapter0.cs]: is this class ever used?

File: Assets/Scripts/UI/Authentication.cs [hunk 4]
```diff
@@ -8,6 +8,7 @@
 using TMPro;
 using UnityEngine.SceneManagement;
 using UnityEngine.UI;
+using Doozy.Engine;
```
Comment (DoubleSentinel) [inline Assets/Scripts/UI/Authentication.cs]: Is this used?

**Review by DoubleSentinel — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Chapters/ControllerChapter0.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        */
+        selectedCharacterIndex = 10; //Otarie as head default
+        UpdateCharacterSelectionUI();
+    }
+    private void Update()
+    {
+        //backgroundColor.color = Color.Lerp(backgroundColor.color, desiredColor, Time.deltaTime * backgroudColorTransitionSpeed);
+    }
+
+    public void LeftArrow()
```
Comment (DoubleSentinel) [inline Assets/Scripts/Chapters/ControllerChapter0.cs]: Have you considered handling circular selection with a modulo?

**Review by DoubleSentinel — CHANGES_REQUESTED** [review]
Few changes here and there you should consider looking at before requesting again

File: Assets/Scripts/Chapters/ControllerChapter0.cs (hunk 1 above)
Comment (ThierryS72) [reply Assets/Scripts/Chapters/ControllerChapter0.cs]: Done!

File: Assets/Scripts/Chapters/ControllerChapter0.cs (hunk 2 above)
Comment (ThierryS72) [reply Assets/Scripts/Chapters/ControllerChapter0.cs]: Changed!

File: Assets/Scripts/Chapters/ControllerChapter0.cs (hunk 3 above)
Comment (ThierryS72) [reply Assets/Scripts/Chapters/ControllerChapter0.cs]: class CharacterSelectObject deleted

File: Assets/Scripts/UI/Authentication.cs (hunk 4 above)
Comment (ThierryS72) [reply Assets/Scripts/UI/Authentication.cs]: using Doozy.Engine canceled

File: Assets/Scripts/Chapters/ControllerChapter0.cs (hunk 5 above)
Comment (ThierryS72) [reply Assets/Scripts/Chapters/ControllerChapter0.cs]: Ok, corrected

**Review by DoubleSentinel — APPROVED** [review]

**Pull request merged**